scripts/get_rackspace_pricing.py

```diff
@@ -1,103 +1,86 @@
+                    else:
+                        infrastructure = prices[flavor_type][flavor_os][flavor_size]['infrastructure']['managed-inf']['USD']
+                        support = prices[flavor_type][flavor_os][flavor_size]['support']['managed-inf']['USD']
+                        sizes[region][size][flavor_os] = infrastructure + support
 
-#FIXME: GBP mapping
-currency_mapping = {
-   'USD': '$',
-   'GBP': 'GBP'
-}
+get_size_per_region(us_prices, region='rackspacenovadfw')
+sizes['rackspacenovaord'] = sizes['rackspacenovadfw']
+sizes['rackspacenovaiad'] = sizes['rackspacenovadfw']
 
-#populate our dict with prices
+get_size_per_region(uk_prices, region='rackspacenovalon')
+get_size_per_region(aus_prices, region='rackspacenovasyd')
+get_size_per_region(apac_prices, region='rackspacenovahkg')
 
-for prod in price_list:
-    description = prod['description'] # description, contains the image type as well
-    for image_type in image_mapping.keys():
-        if image_type in description:
-            image = image_mapping[image_type]
-            break
-    try:
-        for line in prod['productCharacteristics']['productCharacteristic']:
-            if line.get('characteristicCategory') == 'PROVISIONING':
-                size = line['value'] #the size value, values 2-8
-        for pricing in prod['priceList']['productOfferingPrice'][0]['prices']['price']: 
-            currency = currency_mapping.get(pricing['currency'], '')
-            amount = currency + pricing['amount'] + "/hour"
-            region = 'rackspacenova' + pricing['region'].lower()
-            try:
-                rackspace[region][size][image] = amount
-            except:
-                rackspace[region][size] = {}
-                rackspace[region][size][image] = amount
-    except Exception as e: 
-        pass
-        #dicts for Bandwidth/Storage that do not interest us
+# TODO: rackspace first gen http://www.rackspace.com/cloud/pricing/
 
-        
-#formatting for easy copy/paste to mist.io/config.py
-for rack_key in rackspace.keys(): 
-    print "        \"%s\": {" % rack_key
-    for image in rackspace[rack_key].keys()[:-1]:
-        print "            \"%s\": %s," % (image, json.dumps(rackspace[rack_key][image]))
-    image = rackspace[rack_key].keys()[-1]   
-    print "            \"%s\": %s" % (image, json.dumps(rackspace[rack_key][image]))   
-    #don't use a comma for the last key, for valid JSON
-    print '        },\n'
 
+#formatting for easy copy/paste to mist.io/config.py
+for rack_key in sizes.keys():
+    if sizes[rack_key].keys():
+        print "        \"%s\": {" % rack_key
+        for image in sizes[rack_key].keys()[:-1]:
+            print "            \"%s\": %s," % (image, json.dumps(sizes[rack_key][image]))
+        image = sizes[rack_key].keys()[-1]
+        print "            \"%s\": %s" % (image, json.dumps(sizes[rack_key][image]))
+        #don't use a comma for the last key, for valid JSON
+        print '        },\n'
```

Comment:
Isn't there a nicer way to print these? Such as creating a single object and then dumping it to json?